jaeger-jaxrs2/src/test/java/com/uber/jaeger/propagation/ServerFilterIntegrationTest.java

```diff
@@ -0,0 +1,197 @@
+  }
+
+  @Override
+  @After
+  public void tearDown() throws Exception {
+    super.tearDown();
+
+    // stop the server
+    if (server.isStarted()) {
+      server.stop();
+    }
+  }
+
+  @Override
+  protected Application configure() {
+
+    io.opentracing.Tracer serverTracer =
+        new com.uber.jaeger.Tracer
+            .Builder("server-tracer", new NoopReporter(), new ConstSampler(true))
+            .build();
+
+    TraceContext serverTraceContext = new ActiveSpanSourceTraceContext(serverTracer);
+    
+    // Our 'Application'
+    ResourceConfig resourceConfig =
+        new ResourceConfig()
+            .register(new ServerFilter(serverTracer, serverTraceContext), Priorities.USER + 10)
+            .register(new TestHookFilter(serverTraceContext), Priorities.USER + 20)
+            .register(TestResource.class);
+
+    return resourceConfig;
+  }
+
+  @After
+  public void teardownTestingInfra() {
+    // stop server
+    if (server.isStarted()) {
+      server.stop();
+    }
+  }
+
+  @Test
+  public void testTracingUsingInstrumentedClient() {
+    com.uber.jaeger.Span span = (com.uber.jaeger.Span) clientTracer
+        .buildSpan("root-span")
+        .startManual();
+    span.setBaggageItem("x-test", "testing");
+    clientTraceContext.push(span);
+
+    Response resp = target()
+        .register(new ClientFilter(clientTracer, clientTraceContext))
+        .path("test")
+        .request()
+        .header(DEBUG_JAEGER_SPAN_BAGGAGE_HEADER, "foo=bar")
+        .get();
+    assertEquals(resp.getStatus(), 200);
+  }
+
+  @Test
+  public void testTracingUsingInstrumentedClientWithFakeBaggageHeader() {
+    Response resp = target()
+        .register(new ClientFilter(clientTracer, clientTraceContext))
+        .path("test")
+        .request()
+        .header(DEBUG_JAEGER_SPAN_BAGGAGE_HEADER, "foo=bar")
+        .header("uberctx-x-test", "testing")
+        .get();
+    assertEquals(resp.getStatus(), 200);
+  }
+
+  @Test
+  public void testTracingUsingNonInstrumentedClient() {
+    Response resp = target()
+        .path("test")
+        .request()
+        .header(DEBUG_ID_HEADER_KEY, "12345")
+        .header(DEBUG_JAEGER_SPAN_BAGGAGE_HEADER, "foo=bar,x-test=testing")
+        .get();
+    assertEquals(resp.getStatus(), 200);
+  }
+
+  @Test
+  public void testTracingUsingNonInstrumentedClientWithFakeBaggageHeader() {
+    Response resp = target()
+        .path("test")
+        .request()
+        .header(DEBUG_ID_HEADER_KEY, "12345")
+        .header(DEBUG_JAEGER_SPAN_BAGGAGE_HEADER, "foo=bar")
+        .header("uberctx-x-test", "testing")
+        .get();
+    assertEquals(resp.getStatus(), 200);
+  }
+
+  @Test
+  public void testTracingUsingNonInstrumentedClientAndNoDebugId() {
+    /**
+     * this test shows that in case where there is no uber-trace-id (non-instrumented client),
+     * only way to force baggage to get populated is to include the ``"jaeger-debug-id"`` header.
```

Comment:
yes, but why do we want that? Let's come up with a minimum & consistent set of rules of what we want. For example (not necessarily that we MUST do it this way, but at least it has some internal consistency):
  * when trace ID is present
     * baggage from uberctx- is respected
     * baggage from jaeger-baggage is not respected
  * when trace ID is not present
     * baggage from uberctx- is not respected
     * baggage from jaeger-baggage is respected
  * debug-id is always respected regardless of other data

Another alternative is to always respect everything (probably the simplest).